runelite-client/src/main/java/net/runelite/client/plugins/bosstimer/RespawnTimer.java

```diff
@@ -25,22 +25,39 @@
 package net.runelite.client.plugins.bosstimer;
 
 import java.awt.image.BufferedImage;
+import java.time.Duration;
+import java.time.Instant;
 import java.time.temporal.ChronoUnit;
-import net.runelite.client.plugins.Plugin;
 import net.runelite.client.ui.overlay.infobox.Timer;
 
 class RespawnTimer extends Timer
 {
 	private final Boss boss;
+	private final BossTimersPlugin bossPlugin;
+	private final String name;
 
-	public RespawnTimer(Boss boss, BufferedImage bossImage, Plugin plugin)
+	RespawnTimer(Boss boss, BufferedImage bossImage, BossTimersPlugin plugin, String name)
 	{
 		super(boss.getSpawnTime().toMillis(), ChronoUnit.MILLIS, bossImage, plugin);
 		this.boss = boss;
+		this.bossPlugin = plugin;
+		this.name = name;
 	}
 
 	public Boss getBoss()
 	{
 		return boss;
 	}
+
+	@Override
+	public boolean render()
+	{
+		Duration timeLeft = Duration.between(Instant.now(), endTime);
+		if (timeLeft.isNegative())
+		{
+			bossPlugin.notifyRespawn(name);
```

Comment:
Whoops, just went to test it more by enabling the chat notifications and realized that it sends notifications in multiple frames. For some reason I thought once render returns false that the infobox would be removed in the same frame. Will have to change how this works.